src/puppetlabs/trapperkeeper/core.clj

```diff
@@ -35,10 +35,11 @@
   [services config-data]
   {:pre  [(sequential? services)
           (every? #(satisfies? services/ServiceDefinition %) services)
-          (map? config-data)]
+          (ifn? config-data)]
    :post [(satisfies? app/TrapperkeeperApp %)]}
-  (config/initialize-logging! config-data)
-  (internal/build-app* services config-data (promise)))
+  (let [config-data-fn #(if (map? config-data) (constantly config-data) config-data)]
```

Comment:
Also, do we need the anonymous function that wraps the `if map?` or can we just get rid of that like:
```clj
(let [config-data-fn (if (map? config-data) (constantly config-data) config-data] ...)
```